bus-mapping/src/evm/opcodes.rs

```diff
@@ -233,8 +250,27 @@ pub fn gen_associated_ops(
     state: &mut CircuitInputStateRef,
     geth_steps: &[GethExecStep],
 ) -> Result<Vec<ExecStep>, Error> {
-    let fn_gen_associated_ops = fn_gen_associated_ops(opcode_id);
-    fn_gen_associated_ops(state, geth_steps)
+    let opcode = down_cast_to_opcode(opcode_id);
+    if geth_steps.len() > 1 && opcode_id.need_reconstruction() {
+        let memory = opcode.reconstruct_memory(state, geth_steps)?;
+        if geth_steps[1].memory.borrow().is_empty() {
+            geth_steps[1].memory.replace(memory.clone());
+        } else {
+            assert_eq!(&memory, geth_steps[1].memory.borrow().deref());
```

Comment:
add comments here:

line 257 happens when memory trace disabled, while line 259  enabled.
line 259 is for debugging for next several months. In long term it is not needed